### ISSUE TITLE
Having trouble installing on Linux and windows

### ISSUE THREAD

**Issue by Braber01** [boo-lang/boo#197]
every time I try to run the script on each platform I keep getting a bunch of errors and I'm not really sure how to fix them

**Comment by Braber01** [comment]
mkdir -p bootstrap
cp -R lib/ bootstrap/lib
# Mono loads log4net before privatebinpath is set-up, so we need this in the same directory
# as NAnt.exe
cp lib/common/neutral/log4net.dll bootstrap
cp src/NAnt.Console/App.config bootstrap/NAnt.exe.config
/Library/Frameworks/Mono.framework/Versions/Current/bin/mcs  -target:exe -define:MONO,NET_1_0,NET_1_1,NET_2_0,ONLY_2_0 -out:bootstrap/NAnt.exe -r:bootstrap/log4net.dll \
	-r:System.Configuration.dll -recurse:src/NAnt.Console/*.cs src/CommonAssemblyInfo.cs
/bin/sh: 1: /Library/Frameworks/Mono.framework/Versions/Current/bin/mcs: not found
Makefile:98: recipe for target 'bootstrap/NAnt.exe' failed
make: *** [bootstrap/NAnt.exe] Error 127

Here you go

**Comment by masonwheeler** [comment]
All right.  Looks like it's trying to run `mcs`, the Mono C# compiler, at `/Library/Frameworks/Mono.framework/Versions/Current/bin/mcs`, and it can't find it.  Can you verify if that path exists?  If it's not there, you probably don't have Mono installed on your system.

**Comment by masonwheeler** [comment]
All right.  But does that specific path exist?  Or do you have `mcs` installed somewhere else?

I wasn't the one who set up the Linux build scripts, so I don't know exactly how this side of it works, but it sure looks like it's looking for `mcs` at that specific path and not finding it there, even if you do have it somewhere else.

**Comment by Braber01** [comment]
no for some reason the path doesn't exist, and I'm not sure where else it could be.

**Comment by masonwheeler** [comment]
All right, looks like the source of the problem is [here](https://github.com/boo-lang/boo/blob/10cfbf08e0f5568220bc621606a3e49d48dc69a5/build-tools/.config-build#L15).  The `MONO_DIR` environment variable is not defined, and it's plugging that path in as a default.

Not 100% sure, but it looks like you can fix this by finding the Mono install path on your system and setting the `MONO_DIR` environment variable to that path, and then re-running this.

**Comment by Braber01** [comment]
I honestly have no clue where the path is defined it looks like the program is in /usr/bin but Iḿ not sure where everything I need is located

**Comment by masonwheeler** [comment]
@eatdrinksleepcode Can you weigh in on this one?  It's getting out past the edges of my knowledge at this point...

**Comment by Braber01** [comment]
@masonwheeler is there a gitter.im room for the boo programming language? I don't want to keep constantly bothering you about a visual studio extension, I was able to get boo to work on windows with sharpDevlop, but that's about it

**Comment by masonwheeler** [comment]
No, unfortunately.  I've looked into setting one up, but I don't have the access rights to associate an external app with boo-lang.  I think @bamboo is the only one who can do that.

**Comment by Braber01** [comment]
which mcs says /usr/bin/mcs I'm using linux mint if it helps
